Paintroid/src/main/java/org/catrobat/paintroid/dialog/SaveInformationDialog.kt

```diff
@@ -0,0 +1,207 @@
+        imageName = customLayout.findViewById(R.id.pocketpaint_image_name_save_text)
+    }
+
+    private fun initializeFunctioning() {
+        val spinnerArray = arrayListOf("png", "jpg", "ora")
+        val adapter =
+            ArrayAdapter(mySpinner.context, android.R.layout.simple_spinner_item, spinnerArray)
+        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+        mySpinner.adapter = adapter
+        val percentageString = Integer.toString(FileIO.compressQuality)
+        percentage.text = percentageString
+        imageName.text = setName
+        mySeekbar.progress = FileIO.compressQuality
+        mySeekbar.setOnSeekBarChangeListener(this)
+        mySpinner.onItemSelectedListener = this
+        informationButton.setOnClickListener {
+            when {
+                FileIO.isCatrobatImage -> {
+                    presenter.showOraInformationDialog()
+                }
+                FileIO.compressFormat == Bitmap.CompressFormat.JPEG -> {
+                    presenter.showJpgInformationDialog()
+                }
+                else -> {
+                    presenter.showPngInformationDialog()
+                }
+            }
+        }
+    }
+
+    override fun onItemSelected(parent: AdapterView<*>, view: View, position: Int, id: Long) {
+        when (parent.getItemAtPosition(position).toString()) {
+            "jpg" -> {
+                specificFormatLayout.removeAllViews()
+                specificFormatLayout.addView(jpgView)
+                FileIO.compressFormat = Bitmap.CompressFormat.JPEG
+                FileIO.isCatrobatImage = false
+                FileIO.ending = ".jpg"
+            }
+            "png" -> {
+                specificFormatLayout.removeAllViews()
+                FileIO.compressFormat = Bitmap.CompressFormat.PNG
+                FileIO.isCatrobatImage = false
+                FileIO.ending = ".png"
+            }
+            "ora" -> {
+                specificFormatLayout.removeAllViews()
+                FileIO.compressFormat = Bitmap.CompressFormat.PNG
+                FileIO.isCatrobatImage = true
+                FileIO.ending = ".ora"
+            }
+        }
+    }
+
+    override fun onNothingSelected(parent: AdapterView<*>?) = Unit
+    override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
+        val stringToInsert = "$progress%"
+        percentage.text = stringToInsert
+        FileIO.compressQuality = progress
+    }
+
+    override fun onStartTrackingTouch(seekBar: SeekBar) = Unit
+    override fun onStopTrackingTouch(seekBar: SeekBar) = Unit
+
+    companion object {
+        @JvmStatic
+        fun newInstance(
+            permissionCode: Int,
+            imageNumber: Int,
+            isStandard: Boolean,
+            isExport: Boolean
+        ): SaveInformationDialog {
+            if (isStandard) {
+                FileIO.isCatrobatImage = false
```

Comment:
Hmm, looks like static magic. 🤔  Please have a look at FileIO and its usages, maybe it can be refactored to clean code. And create a GSoC ticket :)

Reply:
hmm.. I've not looked at that file, I'll maybe refactor that file later, I'll create the respective ticket then 😄